File: 2448-count-number-of-bad-pairs/2448-count-number-of-bad-pairs.java
class Solution {
    public long countBadPairs(int[] nums) {
        int n = nums.length;
        long tot = (long)n * (n - 1) / 2;
        long good = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < n; i++) {
            int diff = nums[i] - i;
            if (map.containsKey(diff))
                good += (long) map.get(diff);
            map.put(diff, map.getOrDefault(diff, 0) + 1);
        }
        return tot - good;
    }
}

File: 2448-count-number-of-bad-pairs/2448-count-number-of-bad-pairs.py
class Solution(object):
    def countBadPairs(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        tot = n * (n - 1) // 2
        freq = {}
        good = 0
        for i, num in enumerate(nums):
            diff = num - i
            if diff in freq:
                good += freq[diff]
                freq[diff] += 1
            else:
                freq[diff] = 1
        
        return tot - good